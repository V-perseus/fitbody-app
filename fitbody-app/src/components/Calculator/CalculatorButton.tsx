import React from 'react'
import { StyleSheet, TouchableOpacity, Text } from 'react-native'

import globals from '../../config/globals'

interface ICalculatorButtonProps {
  active: boolean
  handlePress: () => void
  title: string
}
const CalculatorButton: React.FC<ICalculatorButtonProps> = (props) => (
  <TouchableOpacity
    style={[styles.button, props.active ? styles.activeButton : styles.inactiveButton]}
    onPress={props.handlePress}
    testID="calc_btn">
    <Text style={props.active ? styles.activeLabel : styles.inactiveLabel}>{props.title}</Text>
  </TouchableOpacity>
)

export default CalculatorButton

const styles = StyleSheet.create({
  button: {
    // height: 20,
    width: globals.window.width * 0.2,
    borderRadius: 50,
    borderColor: globals.styles.colors.colorGray,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  activeButton: {
    backgroundColor: globals.styles.colors.colorPink,
  },
  inactiveButton: {
    backgroundColor: globals.styles.colors.colorWhite,
    borderColor: globals.styles.colors.colorGrayDark,
    borderWidth: 1,
  },
  activeLabel: {
    fontSize: 10,
    color: globals.styles.colors.colorWhite,
  },
  inactiveLabel: {
    fontSize: 10,
    color: globals.styles.colors.colorGrayDark,
  },
})
