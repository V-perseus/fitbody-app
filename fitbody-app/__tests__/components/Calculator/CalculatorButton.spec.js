import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

import CalculatorButton from '../../../src/components/Calculator/CalculatorButton'

describe('CalculatorButton', () => {
  const onPress = jest.fn()

  it('renders snapshot correctly', () => {
    const tree = render(<CalculatorButton onPress={onPress} title="test" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render button text', () => {
    const { queryByText } = render(<CalculatorButton onPress={onPress} title="test" />)
    expect(queryByText('test')).not.toBeNull()
  })

  it('should handle press event', async () => {
    const { getByTestId } = render(<CalculatorButton handlePress={onPress} title="test" />)
    await fireEvent.press(getByTestId('calc_btn'))
    expect(onPress).toHaveBeenCalled()
  })
})
