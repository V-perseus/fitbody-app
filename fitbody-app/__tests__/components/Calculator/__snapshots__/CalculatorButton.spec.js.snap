// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorButton renders snapshot correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "#bbbbbb",
      "borderRadius": 10,
      "borderWidth": 1,
      "flexDirection": "column",
      "height": 20,
      "justifyContent": "center",
      "opacity": 1,
      "width": 150,
    }
  }
  testID="calc_btn"
>
  <Text
    style={
      {
        "color": "#bbbbbb",
        "fontSize": 10,
      }
    }
  >
    test
  </Text>
</View>
`;
